clang/lib/Parse/ParsePragma.cpp

```diff
@@ -1400,6 +1416,171 @@ bool Parser::HandlePragmaMSAllocText(StringRef PragmaName,
   return true;
 }
 
+NestedNameSpecifier *
+Parser::zOSParseIdentifier(StringRef PragmaName,
+                           const IdentifierInfo *IdentName) {
+  NestedNameSpecifier *NestedId = nullptr;
```

Comment:
What is this for the grammar of?  Are you just trying to get the name of a thing?  I did something similar, see : `ParseOpenACCIDExpression`.  For C++ you can use `ParseCXXIdExpression`, and I emulated it for C.